vm/systemSmartContracts/delegation.go

```diff
@@ -839,6 +839,7 @@ func (d *delegation) unBondNodes(args *vmcommon.ContractCallInput) vmcommon.Retu
 	successKeys, _ := getSuccessAndUnSuccessKeys(vmOutput.ReturnData, args.Arguments)
 	for _, successKey := range successKeys {
 		status.UnStakedKeys, status.NotStakedKeys = moveNodeFromList(status.UnStakedKeys, status.NotStakedKeys, successKey)
+		status.StakedKeys, status.NotStakedKeys = moveNodeFromList(status.StakedKeys, status.NotStakedKeys, successKey)
```

Comment:
is there a case when a stakedKey moves directly to notStaked, without being unstaked first?

Reply:
If there is a desynchronization error between the contracts. Only safety measures to not block states.

Reply:
ok